pkg/daemon/daemon.go

```diff
@@ -715,8 +715,7 @@ func (dn *Daemon) syncNodeHypershift(key string) error {
 	}
 	targetHash := string(targetHashBytes)
 
-	// TODO probably have to compress this in the future
-	ignConfig, err := ctrlcommon.ParseAndConvertConfig(ignServedConfigBytes)
+	ignConfig, err := ctrlcommon.ParseAndConvertGzippedConfig(ignServedConfigBytes)
```

Comment:
For my understanding, is this gzip handling only needed in HyperShift case or could be useful for regular cluster as well?

Reply:
I think for now it makes sense to just have it in Hypershift. MachineConfigs don't really have the issue of size and we want them to be readable (and in terms of space saving, we will get more value from deleting stale rendered configs I think)